### PR TITLE
Improved random generator used by OSPRay renderers

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+#include "ospray/math/vec.ih"
+
+/**
+    Returns a random direction based on location in the frame buffer, the normal
+    to the surface and rotation parameters for the precomputed hard-coded random
+    distribution.
+    @param sample Frame buffer sample being rendered
+    @param normal Normal vertor to the surface
+*/
```
Comment (jafyvilla) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: Missing doxygen for param randomNumber?

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+vec3f getRandomVector(
+    varying ScreenSample& sample,
+    const vec3f& normal,
+    const int randomNumber );
```
Comment (jafyvilla) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: No const needed for copy parameters in header

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+    Returns a tangent vectors for a given normal.
```
Comment (jafyvilla) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: Returns two?/the? tangent vectors

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih (hunk 1 above)
Comment (jafyvilla) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: Also missing the @return

**Comment by jafyvilla** [comment]
Other than that, looks good! 👍

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih (hunk 1 above)
Comment (favreau) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: ok. Thanks

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih (hunk 2 above)
Comment (favreau) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: I know, but I still do not agree with that coding style. Anyways, removed! :-)

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih (hunk 3 above)
Comment (favreau) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: Two. Fixed.

File: plugins/renderers/ospray/render/utils/RandomGenerator.ih (hunk 2 above)
Comment (jafyvilla) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: Ah! then up to you! I don't want to influence your opinion ;)

Comment (favreau) [inline plugins/renderers/ospray/render/utils/RandomGenerator.ih]: Well, ISPC decided for us. Linking fails if definition varies between .ispc and .ih. Probably one of the reasons why I don't like this convention ;-) So I put the const back.

**Pull request merged**